packages/jetbrains-plugin/src/test/kotlin/com/mongodb/jbplugin/fixtures/AssertionExtensions.kt

```diff
@@ -21,6 +22,38 @@ fun RemoteRobot.closeAllOpenModals() {
     }
 }
 
+/**
+ * Waits until the block function finishes successfully up to 1 second (or the provided timeout).
+ *
+ * Example usages:
+ *
+ * ```kt
+ * eventually("Doing something") {
+ *    verify(mock).myFunction()
+ * }
+ * // with custom timeout
+ * eventually(description = "Doing something", timeout = Duration.ofSeconds(5)) {
+ *    verify(mock).myFunction()
+ * }
+ * ```
+ *
+ * @param timeout
+ * @param fn
+ * @param recovery
+ */
+fun eventually(
+    description: String,
+    timeout: Duration = Duration.ofSeconds(1),
+    recovery: () -> Unit = {},
+    fn: (Int) -> Unit
+) {
+    eventually(timeout, recovery) { attempt ->
+        step("$description, attempt=$attempt") {
+            fn(attempt)
+        }
+    }
+}
```

Comment:
Added this to address the suggestion about combining eventually and step descriptions.